lang/python/pillow/patches/0001-setup.py-use-patch-from-Debian-and-tweak-it.patch

```diff
@@ -0,0 +1,316 @@
++            feature.zlib = "z"
+ 
+         if feature.want('jpeg'):
+-            _dbg('Looking for jpeg')
+-            if _find_include_file(self, "jpeglib.h"):
+-                if _find_library_file(self, "jpeg"):
+-                    feature.jpeg = "jpeg"
+-                elif (sys.platform == "win32" and
+-                      _find_library_file(self, "libjpeg")):
+-                    feature.jpeg = "libjpeg"  # alternative name
++            feature.jpeg = "jpeg"
+ 
+         feature.openjpeg_version = None
+         if feature.want('jpeg2000'):
+@@ -537,13 +457,7 @@ class pil_build_ext(build_ext):
+                     feature.imagequant = "libimagequant"
+ 
+         if feature.want('tiff'):
+-            _dbg('Looking for tiff')
+-            if _find_include_file(self, 'tiff.h'):
+-                if _find_library_file(self, "tiff"):
+-                    feature.tiff = "tiff"
+-                if (sys.platform in ["win32", "darwin"] and
+-                        _find_library_file(self, "libtiff")):
+-                    feature.tiff = "libtiff"
++            feature.tiff= "tiff"
+ 
+         if feature.want('freetype'):
+             _dbg('Looking for freetype')
+@@ -623,7 +537,7 @@ class pil_build_ext(build_ext):
+         if feature.jpeg2000:
+             libs.append(feature.jpeg2000)
+             defs.append(("HAVE_OPENJPEG", None))
+-            if sys.platform == "win32":
++            if host_platform == "win32":
+                 defs.append(("OPJ_STATIC", None))
+         if feature.zlib:
+             libs.append(feature.zlib)
+@@ -634,12 +548,12 @@ class pil_build_ext(build_ext):
+         if feature.tiff:
+             libs.append(feature.tiff)
+             defs.append(("HAVE_LIBTIFF", None))
+-        if sys.platform == "win32":
++        if host_platform == "win32":
+             libs.extend(["kernel32", "user32", "gdi32"])
+         if struct.unpack("h", "\0\1".encode('ascii'))[0] == 1:
+             defs.append(("WORDS_BIGENDIAN", None))
+ 
+-        if sys.platform == "win32" and not (PLATFORM_PYPY or PLATFORM_MINGW):
++        if host_platform == "win32" and not (PLATFORM_PYPY or PLATFORM_MINGW):
+             defs.append(("PILLOW_VERSION", '"\\"%s\\""' % PILLOW_VERSION))
+         else:
+             defs.append(("PILLOW_VERSION", '"%s"' % PILLOW_VERSION))
+@@ -661,7 +575,7 @@ class pil_build_ext(build_ext):
+ 
+         if feature.lcms:
+             extra = []
+-            if sys.platform == "win32":
++            if host_platform == "win32":
+                 extra.extend(["user32", "gdi32"])
+             exts.append(Extension("PIL._imagingcms",
+                                   ["src/_imagingcms.c"],
+@@ -681,7 +595,7 @@ class pil_build_ext(build_ext):
+                                   libraries=libs,
+                                   define_macros=defs))
+ 
+-        tk_libs = ['psapi'] if sys.platform == 'win32' else []
++        tk_libs = ['psapi'] if host_platform == 'win32' else []
```

Comment:
why does this need to change ?
or is it needed ?
not sure if it will be true, ever ?

Reply:
I'm not really sure why the original Debian patch replaces `sys.platform` with `host_platform` (set from `sysconfig.get_platform()`), but if I had to guess I would say for cross-compilation reasons.

I think we should probably continue replacing `sys.platform` with `host_platform` as well, but instead set `host_platform = "linux"`. In theory, this could be compiling on macOS where `sys.platform` (for host Python) would be `"darwin" (with similar changes on other host OSes).